File: lib/languages/typescript/modules.ts
// File: math.ts
// Exported functions
export function add(a: number, b: number): number {
  return a + b
}

export function subtract(a: number, b: number): number {
  return a - b
}

// Private function (not exported)
function square(x: number): number {
  return x * x
}

// File: main.ts
import { add } from "./math"
// or import * as math from './math';

function main() {
  const result = add(5, 3) // 8
  // square(4);  // Error: square is not exported
}

File: lib/languages/kotlin/modules.kt
// File: Math.kt
package com.example.math

// Public functions are exported
fun add(a: Int, b: Int): Int {
   return a + b
}

fun subtract(a: Int, b: Int): Int {
   return a - b
}

// Private function (not exported outside the file)
private fun square(x: Int): Int {
   return x * x
}

// File: Main.kt
package com.example.main

import com.example.math.add
// or import com.example.math.*

fun main() {
   val result = add(5, 3)  // 8
   // square(4)  // Error: square is private
}

